src/current/v22.2/upgrade-cockroach-version.md

```diff
@@ -256,10 +256,27 @@ To roll back an upgrade, do the following on each cluster node:
 ## Step 6. Finish the upgrade
 
 {{site.data.alerts.callout_info}}
-This step is relevant only when upgrading from {{ previous_version }}.x to {{ page.version.version }}. For upgrades within the {{ page.version.version }}.x series, skip this step.
+Finalization is required only when upgrading from {{ previous_version }}.x to {{ page.version.version }}. For upgrades within the {{ page.version.version }}.x series, skip this step.
 {{site.data.alerts.end}}
 
-If you disabled auto-finalization in [step 3](#step-3-decide-how-the-upgrade-will-be-finalized), monitor the stability and performance of your cluster for as long as you require to feel comfortable with the upgrade (generally at least a day). If during this time you decide to roll back the upgrade, repeat the rolling restart procedure with the previous binary.
+If you disabled auto-finalization in [step 3](#step-3-decide-how-the-upgrade-will-be-finalized), monitor the stability and performance of your cluster for as long as you require to feel comfortable with the upgrade (generally at least a day). If during this time you decide to roll back the upgrade, repeat the rolling restart procedure with the previous binary. Otherwise, you must re-enable upgrade finalization to complete the upgrade to {{ page.major_version }}. Cockroach Labs recommends that you either finalize or roll back a major-version upgrade within a relative short period of time; running in a partially-upgraded state is not recommended.
+
+{{site.data.alerts.callout_danger}}
+A cluster that is not finalized on {{ previous_version }} cannot be upgraded to {{ page.major_version }} until the {{ previous_version }} upgrade is finalized.
+{{site.data.alerts.end}}
+
+When finalization begins, a series of migration jobs run to enable certain types of features and changes in the new major version that cannot be rolled back. These include changes to system schemas, indexes, and descriptors, and enabling certain types of improvements and new features. Until the upgrade is finalized, these features and functions will not be available and the command `SHOW CLUSTER SETTING version` will return `{{ previous_version }}`.
+
+{% capture previous_version_numeric %}{{ previous_version | strip_first: 'v'}}{% endcapture %}
+{% capture major_version_numeric %}{{ page.major_version | strip_first: 'v' }}{% endcapture %}
+
+You can monitor the process of the migration in the DB Console [Jobs page]({% link {{ page.version.version }}/ui-jobs-page.md %}). Migration jobs have names in the format `{{ major_version_numeric }}-{migration-id}`. If a migration job fails or stalls, Cockroach Labs can use the migration ID to help diagnose and troubleshoot the problem. Each major version has different migration jobs with different IDs.
```

Comment:
Is it possible to integrate this and the paragraphs below with the numbered steps? I think it is confusing to have several paragraphs of prescriptive text preceding another series of numbered steps. It looks like these paragraphs would apply between steps 2 and 3.

Reply:
Actually, after you finish step 4, you have to make an active decision that puts you into step 5 (Roll back the upgrade) or step 6 (Finish the upgrade). Otherwise you are stuck in the unfinalized state where you can't access the new features or do a subsequent major-version upgrade. If you are doing step 6, the only way to know that finalization has finished is to query for the version, and the only way to know its progress if it hasn't yet finished is to do line 273. Does that make sense?

Reply:
Good idea!